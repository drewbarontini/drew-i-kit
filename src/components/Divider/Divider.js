import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ifProp } from 'styled-tools';

import { setProp, palette } from '../../lib/utils';

const Divider = styled.div`
  ${setProp({
    prop: 'color',
    cssProp: 'background-color',
    themeKey: 'colors',
    fallback: props => palette('subdued')(props),
  })};
  ${setProp({
    prop: 'size',
    cssProp: props => ifProp('vertical', 'width', 'height')(props),
    fallback: '1px',
  })};
  ${setProp({
    prop: 'marginBottom',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'marginTop',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: props => ifProp('vertical', 'height', 'width')(props),
    fallback: '100%',
  })};
`;

Divider.propTypes = {
  color: PropTypes.string,
  marginBottom: PropTypes.string,
  marginTop: PropTypes.string,
  size: PropTypes.string,
  vertical: PropTypes.bool,
  width: PropTypes.string,
};

Divider.displayName = 'Divider';

export default Divider;
