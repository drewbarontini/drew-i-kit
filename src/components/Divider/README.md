# Divider

## Usage

```jsx
import { Divider } from 'drew-i-kit';

<Divider />
```

```jsx
import { Divider } from 'drew-i-kit';

<Divider vertical height="50px" />
<Divider size="4px" />
<Divider size="4px" color="primary.base" />
<Divider size="4px" color="primary.base" width="50px" />
```

## Props

- `color` (**string**, optional)
- `marginBottom` (**string**, optional)
- `marginTop` (**string**, optional)
- `size` (**string**, optional)
- `vertical` (**Boolean**, optional)
- `width` (**string**, optional)