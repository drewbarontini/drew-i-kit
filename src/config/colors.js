import { paletteFor } from '../lib';

const colors = {
  ...paletteFor({
    area: 'primary',
    base: '#3576a7',
    darker: ['#152f43', '#1d415c', '#255375', '#2d648e'],
    lighter: ['#538bb4', '#729fc1', '#90B4cf', '#aeC8dc'],
  }),
  ...paletteFor({
    area: 'neutral',
    base: '#737373',
    darker: ['#0b0b0b', '#1d1d1d', '#2e2e2e', '#3f3f3f', '#515151', '#626262'],
    lighter: ['#888888', '#9d9d9d', '#b2b2b2', '#c7c7c7', '#dcdcdc', '#f1f1f1'],
  }),
  accent: {
    ...paletteFor({
      area: 'green',
      base: '#8acb88',
      darker: ['#233322', '#375136', '#4c704b', '#618e5f', '#75ad74'],
      lighter: ['#9cd39a', '#addbac', '#bfe2be', '#d0eacf', '#e2f2e1'],
    }),
    ...paletteFor({
      area: 'yellow',
      base: '#ffc527',
      darker: ['#40310a', '#664f10', '#8c6c15', '#b38a1b', '#d9a721'],
      lighter: ['#ffce47', '#ffd668', '#ffdf88', '#ffe8a9', '#fff1c9'],
    }),
    ...paletteFor({
      area: 'red',
      base: '#fe5f55',
      darker: ['#401815', '#662622', '#8c342f', '#b2433b', '#d85148'],
      lighter: ['#fe776f', '#fe8f88', '#fea7a2', '#ffbfbb', '#ffd7d5'],
    }),
  },
  bg: '#fff',
  fg: '#272727',
  white: '#fff',
  black: '#272727',
  border: '#dcdcdc',
};

export default colors;
