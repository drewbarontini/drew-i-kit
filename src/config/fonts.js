const fonts = {
  family: {
    base:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    heading:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    monospace: "'Monaco', monospace",
  },
  lineHeight: {
    base: '1.5',
    heading: '1.2',
  },
  sizes: {
    base: '14px',
    xxs: '8px',
    xs: '10px',
    s: '12px',
    l: '16px',
    xl: '24px',
    h1: '30px',
    h2: '26px',
    h3: '22px',
    h4: '18px',
  },
  weights: {
    light: 200,
    normal: 400,
    semiBold: 600,
    bold: 800,
  },
};

export default fonts;
