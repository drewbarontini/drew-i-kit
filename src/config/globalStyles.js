import { css } from 'styled-components';

import { palette, theme } from '../lib/props';
import { responsive } from '../lib/responsive';

const globalStyles = css`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  body {
    background-color: ${palette('bg')};
    color: ${palette('fg')};
    font-family: ${theme('fonts.family.base')};
    font-size: ${theme('fonts.sizes.base')};
    line-height: ${theme('fonts.lineHeight.base')};
    margin: ${theme('spacing.l')};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${theme('fonts.family.heading')};
    line-height: ${theme('fonts.lineHeight.heading')};
    margin: 0;
    margin-bottom: ${theme('spacing.s')};
  }

  img {
    display: block;
    max-width: 100%;
  }

  ${responsive};
`;

export default globalStyles;
