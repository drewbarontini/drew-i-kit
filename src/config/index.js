export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
