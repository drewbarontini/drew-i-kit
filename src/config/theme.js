const theme = {
  borders: {
    sizes: {
      base: '1px',
    },
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
    white: '#fff',
    black: '#272727',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
