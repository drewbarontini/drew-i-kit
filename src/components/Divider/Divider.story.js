import React from 'react';
import { storiesOf } from '@storybook/react';

import Divider from './';

storiesOf('Divider', module)
  .add('default', () => <Divider />)
  .add('vertical', () => <Divider vertical height="50px" />)
  .add('size', () => <Divider size="4px" />)
  .add('color', () => <Divider size="4px" color="primary" />)
  .add('width', () => <Divider size="4px" color="primary" width="50px" />);
