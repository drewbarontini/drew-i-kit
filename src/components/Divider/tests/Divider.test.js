import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Divider from '..';

describe('<Divider />', () => {
  it('should render properly', () => {
    const { container } = render(<Divider theme={theme} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName.toLowerCase()).toEqual('div');
    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      theme.colors.border
    );
    expect(container.firstChild).toHaveStyleRule('height', '1px');
    expect(container.firstChild).toHaveStyleRule('width', '100%');
  });
});
