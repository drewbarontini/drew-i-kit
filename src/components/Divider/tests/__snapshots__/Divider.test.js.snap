// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render properly 1`] = `
.c0 {
  background-color: #cbcbcb;
  height: 1px;
  width: 100%;
}

<div
  class="c0"
/>
`;
